app/vmagent/remotewrite/client.go

```diff
@@ -511,6 +531,28 @@ func getRetryDuration(retryAfterDuration, retryDuration, maxRetryDuration time.D
 	return retryDuration
 }
 
+func mustRepackBlockFromZstdToSnappy(zstdBlock []byte) []byte {
+	plainBlock := make([]byte, 0, len(zstdBlock)*2)
+	plainBlock, err := zstd.Decompress(plainBlock, zstdBlock)
+	if err != nil {
+		logger.Panicf("BUG: cannot re-pack block with size %d bytes from Zstd to Snappy: %s", len(zstdBlock), err)
```

Comment:
I'd log this panic with FATAL prefix, since it may occur because of corrupted data on disk